addons/hr_recruitment/models/hr_recruitment.py

```diff
@@ -315,7 +317,23 @@ def create(self, vals):
             vals['date_open'] = fields.Datetime.now()
         if vals.get('email_from'):
             vals['email_from'] = vals['email_from'].strip()
-        return super(Applicant, self).create(vals)
+        res = super().create(vals)
+        # Record creation through calendar, creates the calendar event directly, it will also create the activity.
+        if 'default_activity_date_deadline' in self.env.context:
+            deadline = fields.Datetime.to_datetime(self.env.context.get('default_activity_date_deadline'))
+            partners = res.partner_id | res.user_id.partner_id | res.department_id.manager_id.user_id.partner_id
+
+            category = self.env.ref('hr_recruitment.categ_meet_interview')
+            self.env['calendar.event'].sudo().with_context(default_applicant_id=res.id).create({
+                'applicant_id': res.id,
+                'partner_ids': [(6, 0, partners.ids)],
+                'user_id': self.env.uid,
+                'name': res.name,
+                'categ_ids': category and [category.id] or False,
```

Comment:
`category` will always be the record here, because `ref()` will raise if the record is not found

Reply:
True